msbuild/Xamarin.Shared/Xamarin.Shared.targets

```
@@ -491,6 +491,32 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 
 	<Target Name="Codesign" Condition="'$(_CanOutputAppBundle)' == 'true'" DependsOnTargets="$(CodesignDependsOn)" />
 
+	<Target Name="_ExtendAppExtensionReferences" DependsOnTargets="_ResolveAppExtensionReferences" Condition=" '@(AdditionalAppExtensions)' != ''">
+	<!-- The idea here is that after _ResolveAppExtensionReferences we inject the 3rd party extensions into the list being processed later for embedding and code signing.
+		- _ResolvedAppExtensionReferences is an item group of the path, so that's easy.
+		- _AppExtensionCodesignProperties less so. It is generated by reading codesign.items generated by the c# tasks during build, which we don't have. 
+		We also don't have a concrete list of types that get serialized, so I dug for every property referenced and set what I believe are sane values.-->
+		<PropertyGroup>
+			<_ExtensionEntitlementPath>%(AdditionalAppExtensions.Identity)/%(AdditionalAppExtensions.Name).entitlements</_ExtensionEntitlementPath>
+		</PropertyGroup>
+		<ItemGroup>
+			<_ResolvedAppExtensionReferences Include="%(AdditionalAppExtensions.Identity)/%(AdditionalAppExtensions.BuildOutput)/%(AdditionalAppExtensions.Name).appex" />
+
+			<_AppExtensionCodesignProperties Include="%(AdditionalAppExtensions.Name).appex">
+				<Entitlements Condition="Exists('$(_ExtensionEntitlementPath)')">
+					$(_ExtensionEntitlementPath)
+				</Entitlements>
+				<SigningKey>$(CodesignKey)</SigningKey>
+				<Keychain>$(CodesignKeychain)</Keychain>
+				<DisableTimestamp>False</DisableTimestamp>
+				<DisableTimestamp Condition="'$(_SdkIsSimulator)' == 'true' Or '$(MtouchDebug)' == 'true'">True</DisableTimestamp>
```

This works for both XI and XM:

```suggestion
				<DisableTimestamp Condition="'$(_SdkIsSimulator)' == 'true' Or '$(_BundlerDebug)' == 'true'">True</DisableTimestamp>
```